internal/core/src/segcore/InsertRecord.h

```diff
@@ -347,9 +347,36 @@ struct InsertRecord {
     InsertRecord(
         const Schema& schema,
         const int64_t size_per_chunk,
-        const storage::MmapChunkDescriptorPtr mmap_descriptor = nullptr)
-        : timestamps_(size_per_chunk), mmap_descriptor_(mmap_descriptor) {
+        const storage::MmapChunkDescriptorPtr mmap_descriptor = nullptr,
+        const milvus::storage::MmapConfig mmap_config =
+            storage::MmapManager::GetInstance().GetMmapConfig())
+        : timestamps_(size_per_chunk) {
         std::optional<FieldId> pk_field_id = schema.get_primary_field_id();
+        // todo: @cqy123456, scalar and vector mmap should be depend on GetScalarFieldEnableMmap()/ GetVectorFieldEnableMmap()
+        storage::MmapChunkDescriptorPtr scalar_mmap_descriptor =
+            mmap_config.GetEnableGrowingMmap() ? mmap_descriptor : nullptr;
+        storage::MmapChunkDescriptorPtr vec_mmap_descriptor =
+            mmap_config.GetEnableGrowingMmap() ? mmap_descriptor : nullptr;
+        storage::MmapChunkDescriptorPtr dense_vec_mmap_descriptor = nullptr;
+        // todo: @cqy123456, remove all condition and select of dense_vec_mmap_descriptor later
+        {
+            const auto& segcore_config =
+                milvus::segcore::SegcoreConfig::default_config();
+            auto enable_intermin_index =
+                segcore_config.get_enable_interim_segment_index();
+            auto dense_vec_intermin_index_type =
+                segcore_config.get_dense_vector_intermin_index_type();
+            auto enable_mmap_field_mmap =
+                mmap_config.GetVectorFieldEnableMmap();
+            bool enable_dense_vec_mmap =
+                enable_intermin_index && enable_mmap_field_mmap &&
+                (dense_vec_intermin_index_type ==
```

Comment:
consider using knowhere's util to judge instead of directly using index_type

Reply:
We miss build config or load config in InsertRecord init function to generate `IndexStaticFaced::HasRawData(const` knowhere::IndexType& indexType, const knowhere::IndexVersion& version, const knowhere::Json& `params);`.
And line 362 - 379 will be removed soon.